src/validators/index.js

```diff
@@ -0,0 +1,42 @@
+const PropTypes = require('prop-types')
+
+//The guid validator checks the length of the prop since the guid format is consistent.
+//I can't think of a situation where we're trying to pass through a incorrect prop
+//that's a string of 36 characters.
+
+function chainedGuidLengthChecker(isRequired) {
+  return function(props, propName, componentName) {
+    const guid = props[propName] || ''
+
+    if (!guid && isRequired) {
+      return new Error(
+        `${propName} was empty. This is a required prop for ${componentName}`,
+      )
+    } else if (guid.length !== 36) {
+      return new Error(
+        `${props[propName]} supplied to ${componentName}, expected a guid.`,
+      )
+    }
+  }
+}
+
+let guid = chainedGuidLengthChecker(false)
+guid.isRequired = chainedGuidLengthChecker(true)
+
+//Node appears as a prop in most of the components. I believe this is created by
+//d3 and is not a ReactNode, so it needs a custom PropType
```

Comment:
Precisely. It's originally created by d3 and I think that we decorate it with that `guid` so that it's easily trackable. I'm open to better names. Could it be a `storyPoint`?